File: Exercism/typescript/triangle/triangle.ts
export class Triangle {
  x: number = 0
  y: number = 0
  z: number = 0

  constructor(_x: number, _y: number, _z: number) {
    this.x = _x;
    this.y = _y;
    this.z = _z;
  }

  get isEquilateral() {
    // https://en.wikipedia.org/wiki/Triangle_inequality#Euclidean_geometry
    if(!(Math.max(this.x, this.y, this.z) < this.x + this.y + this.z - Math.max(this.x, this.y, this.z))) {
      return false;
    }

    if(this.x == this.y && this.z == this.y && this.x == this.z) {
      return true;
    }
    else
    {
      return false;
    }
  }

  get isIsosceles() {
    // https://en.wikipedia.org/wiki/Triangle_inequality#Euclidean_geometry
    if(!(Math.max(this.x, this.y, this.z) < this.x + this.y + this.z - Math.max(this.x, this.y, this.z))) {
      return false;
    }

    if(
      this.x == this.y || this.y == this.z || this.x == this.z ||
      (this.x < this.y && this.z < this.y && this.x < this.z)) {
      return true;
    }
    else
    {
      return false;
    }
  }

  get isScalene() {
    // https://en.wikipedia.org/wiki/Triangle_inequality#Euclidean_geometry
    if(!(Math.max(this.x, this.y, this.z) < this.x + this.y + this.z - Math.max(this.x, this.y, this.z))) {
      return false;
    }

    if(this.x != this.y && this.y != this.z && this.y != this.x) {
      return true;
    }
    else
    {
      return false;
    }
  }
}

File: Exercism/csharp/triangle/Triangle.cs
using System;

public static class Triangle
{
    public static bool IsScalene(double side1, double side2, double side3)
    {
        // https://en.wikipedia.org/wiki/Triangle_inequality#Euclidean_geometry
        if(!(Math.Max(side3, Math.Max(side1, side2)) < side1 + side2 + side3 - Math.Max(side3, Math.Max(side1, side2))))
        {
            return false;
        }

        if(side1 != side2 && side2 != side3 && side1 != side3)
        {
            return true;
        }
        else
        {
            return false;
        }
    }

    public static bool IsIsosceles(double side1, double side2, double side3) 
    {
        // https://en.wikipedia.org/wiki/Triangle_inequality#Euclidean_geometry
        if(!(Math.Max(side3, Math.Max(side1, side2)) < side1 + side2 + side3 - Math.Max(side3, Math.Max(side1, side2))))
        {
        return false;
        }

        if(
            side1 == side2 || side2 == side3 || side1 == side3 ||
            (side1 < side2 && side3 < side2 && side1 < side3))
        {
            return true;
        }
        else
        {
            return false;
        }
    }

    public static bool IsEquilateral(double side1, double side2, double side3) 
    {
        // https://en.wikipedia.org/wiki/Triangle_inequality#Euclidean_geometry
        if(!(Math.Max(side3, Math.Max(side1, side2)) < side1 + side2 + side3 - Math.Max(side3, Math.Max(side1, side2))))
        {
            return false;
        }

        if(side1 == side2 && side3 == side2 && side1 == side3)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}
